packages/lib/apps/getEnabledApps.ts

```diff
@@ -2,12 +2,46 @@ import getApps from "@calcom/app-store/utils";
 import type { CredentialData } from "@calcom/app-store/utils";
 import { prisma } from "@calcom/prisma";
 
-const getEnabledApps = async (userCredentials: CredentialData[]) => {
+import type { Prisma } from ".prisma/client";
+
+/**
+ *
+ * @param credentials - Can be user or team credentials
+ * @param filterOnCredentials - Only include apps where credentials are present
+ * @returns A list of enabled app metadata & credentials tied to them
+ */
+const getEnabledApps = async (credentials: CredentialData[], filterOnCredentials?: boolean) => {
+  const filterOnIds = {
+    credentials: {
+      some: {
+        OR: [] as Prisma.CredentialWhereInput[],
+      },
+    },
+  } satisfies Prisma.AppWhereInput;
+
+  if (filterOnCredentials) {
+    const userIds: number[] = [],
+      teamIds: number[] = [];
+
+    for (const credential of credentials) {
+      if (credential.userId) userIds.push(credential.userId);
+      if (credential.teamId) teamIds.push(credential.teamId);
+    }
+    if (userIds.length) filterOnIds.credentials.some.OR.push({ userId: { in: userIds } });
+    if (teamIds.length) filterOnIds.credentials.some.OR.push({ teamId: { in: teamIds } });
+  }
+
   const enabledApps = await prisma.app.findMany({
-    where: { enabled: true },
+    where: {
+      OR: [
+        { enabled: true, ...(filterOnIds.credentials.some.OR.length && filterOnIds) },
+        // Even if filtering on credentials, everyone has Daily installed
+        { slug: "daily-video", enabled: true },
```

Comment:
We shouldn't hardcode Daily here. The approach would be to get all apps with `isGlobal` `true` because those are considered pre-installed.

There are 2 apps now that satisfy this criteria.
<img width="609" alt="Screenshot 2023-07-27 at 2 23 07 PM" src="https://github.com/calcom/cal.com/assets/1780212/3fc1671f-5b41-43df-9092-e69511c44a94">